materials/property.py

```diff
@@ -0,0 +1,135 @@
+"""classes for representing and querying properties of a material."""
+import numpy as np
+import scipy.interpolate
+
+class Property:
+    """A property of a material."""
+    def __init__(self, name, yaml_dict):
+        self.name = name
+        self.default_value = yaml_dict['default_value']
+        self.units = yaml_dict['units']
+        self.reference = yaml_dict['reference']
+
+    def query_value(self):
+        return self.default_value
+
+
+class StateDependentProperty(Property):
+    """A property of a material which depends on state (e.g. temperature)."""
+    def __init__(self, name, yaml_dict):
+        Property.__init__(self, name, yaml_dict)
+        self.state_vars = yaml_dict['variation_with_state']['state_vars']
+        self.state_vars_units = yaml_dict['variation_with_state']['state_vars_units']
+        self._value_type = yaml_dict['variation_with_state']['value_type']
+        self._representation = yaml_dict['variation_with_state']['representation']
+        if 'state_vars_interp_scales' in yaml_dict['variation_with_state']:
+            self._state_vars_interp_scales = yaml_dict['variation_with_state']['state_vars_interp_scales']
+        else:
+            self._state_vars_interp_scales = ['linear'] * len(self.state_vars)
+
+        if self._representation == 'table':
+            if len(self.state_vars) == 1:
+                self._interp_points = np.array(
+                    yaml_dict['variation_with_state'][self.state_vars[0]])
+                self._interp_values = np.array(
+                    yaml_dict['variation_with_state']['values'])
+                if self._state_vars_interp_scales[0] == 'log':
+                    self._interp_points = np.log(self._interp_points)
+            elif len(self.state_vars) == 2:
+                # TODO there has got to be a less awful way to represent a 2+ dimensional
```

Comment:
Oh my

Reply:
yeah this section is complicated. I spent a few hours on it when I was first devising it, but I could not come up with a more elegant solution